src/main/kotlin/uk/gov/gdx/datashare/services/AcquirerEventProcessor.kt

```diff
@@ -23,32 +25,29 @@ class AcquirerEventProcessor(
   private val acquirerEventAuditService: AcquirerEventAuditService,
   private val outboundEventQueueService: OutboundEventQueueService,
 ) {
-
   @JmsListener(destination = "acquirerevent", containerFactory = "awsQueueContainerFactoryProxy")
   @Transactional
-  fun onAcquirerEvent(message: String) {
-    val acquirerEvent = objectMapper.readValue(message, AcquirerEvent::class.java)
+  fun onAcquirerEvent(acquirerEventId: String) {
+    val acquirerEvent = acquirerEventRepository.findByIdOrNull(UUID.fromString(acquirerEventId))
+      ?: throw EventNotFoundException("No acquirer event found with ID $acquirerEventId")
+
     val acquirerSubscription = acquirerSubscriptionRepository.findByIdOrNull(acquirerEvent.acquirerSubscriptionId)
       ?: throw AcquirerSubscriptionNotFoundException(acquirerEvent.acquirerSubscriptionId)
 
-    if (acquirerSubscription.queueName != null) {
-      persistEventAndPushToQueue(acquirerEvent, acquirerSubscription)
-    } else {
-      persistEvent(acquirerEvent)
+    if (acquirerSubscription.queueName == null) {
+      return
```

Comment:
Love an early return, though curious as to why this can be null/could we just filter it out in the acquirerSubscription call?